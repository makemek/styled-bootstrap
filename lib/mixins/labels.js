import { css } from 'styled-components'
import Color from 'color'

export const label_variant = color => css`
  background-color: ${color};

  &[href] {
    &:hover,
    &:focus {
      background-color: ${Color(color).darken(0.1).rgb().string()};
    }
  }
`
