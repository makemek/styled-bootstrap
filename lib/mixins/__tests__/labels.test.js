import * as label from '../labels'
import { shallow } from 'enzyme'
import styled from 'styled-components'
import 'jest-styled-components'
import React from 'react'

describe('mixin: label', () => {
  describe('label_variant', () => {
    it('should accept named color', () => {
      const Comp = styled.span`
        ${label.label_variant('white')}
      `
      const wrapper = shallow(<Comp>styled label</Comp>)
      expect(wrapper).toHaveStyleRule('background-color', 'white')
      expect(wrapper).toMatchSnapshot()
    })

    it('should accept web color', () => {
      const Comp = styled.span`
        ${label.label_variant('#fff')}
      `
      const wrapper = shallow(<Comp>styled label</Comp>)
      expect(wrapper).toHaveStyleRule('background-color', '#fff')
      expect(wrapper).toMatchSnapshot()
    })

    it('should accept hsl color', () => {
      const hsl = 'hsl(0,0%,100%)'
      const Comp = styled.span`
        ${label.label_variant(hsl)}
      `
      const wrapper = shallow(<Comp>styled label</Comp>)
      expect(wrapper).toHaveStyleRule('background-color', hsl)
      expect(wrapper).toMatchSnapshot()
    })
  })
})
