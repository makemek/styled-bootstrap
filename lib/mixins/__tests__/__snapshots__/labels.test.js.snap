// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mixin: label label_variant should accept hsl color 1`] = `
.c0 {
  background-color: hsl(0,0%,100%);
}

.c0[href]:hover,
.c0[href]:focus {
  background-color: rgb(230,230,230);
}

<span
  className="c0"
>
  styled label
</span>
`;

exports[`mixin: label label_variant should accept named color 1`] = `
.c0 {
  background-color: white;
}

.c0[href]:hover,
.c0[href]:focus {
  background-color: rgb(230,230,230);
}

<span
  className="c0"
>
  styled label
</span>
`;

exports[`mixin: label label_variant should accept web color 1`] = `
.c0 {
  background-color: #fff;
}

.c0[href]:hover,
.c0[href]:focus {
  background-color: rgb(230,230,230);
}

<span
  className="c0"
>
  styled label
</span>
`;
